### PR TITLE
Exception logging improvements and other fixes

### PR DESCRIPTION
**Pull request by dgelessus**
* Moved `logStackTrace` from `Settings` to `LogManager` and made it `static`
* Replaced most uses of `Throwable.printStackTrace` with `LogManager.logStackTrace` (except for some in the logging and console internals, which really shouldn't produce more log messages if logging fails)
* Improved some exception handlers - replaced some Pokémon (catch 'em all) exception handlers with specific exceptions, fixed `InterruptedException` handling, added `finally` blocks for some streams
* Fixed a Java 6 compatibility issue in `LegacyMCLauncher`
* Fixed a NPE in `EditModsDialog`

The changes related to `logStackTrace` affected a bunch of files, you may want to look at the commits individually so you can see what's important and what is just a rename.

### CODE REVIEW

File: src/main/java/com/atlauncher/mclauncher/LegacyMCLauncher.java [hunk 1]
```diff
@@ -407,12 +407,6 @@ public static void main(String[] args) {
                 } catch (InstantiationException e) {
                     System.out.println("Applet wrapper failed! Falling back " + "to compatibility mode.");
                     mc.getMethod("main", String[].class).invoke(null, (Object) mcArgs);
-                } finally {
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/mclauncher/LegacyMCLauncher.java]: Not sure what removing this achieves?

File: src/main/java/com/atlauncher/gui/dialogs/EditModsDialog.java [hunk 2]
```diff
@@ -133,26 +133,29 @@ public void windowClosing(WindowEvent arg0) {
...
                     usesCoreMods = App.settings.getMinecraftVersion(instance.getMinecraftVersion()).usesCoreMods();
                 } catch (InvalidMinecraftVersion e1) {
                     LogManager.logStackTrace(e1);
                 }
-                if(usesCoreMods) {
-                    fcd = new FileChooserDialog(Language.INSTANCE.localize("instance.addmod"), Language
-                            .INSTANCE.localize("common.mod"), Language.INSTANCE.localize("common.add"), Language.INSTANCE
-                            .localize("instance.typeofmod"), new String[]{"Mods Folder", "Inside Minecraft.jar",
-                            "CoreMods Mod", "Texture Pack", "Shader Pack"}, new String[]{"jar", "zip", "litemod"});
+                String[] modTypes;
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/gui/dialogs/EditModsDialog.java]: 👍

File: src/main/java/com/atlauncher/gui/dialogs/EditModsDialog.java [hunk 3]
```diff
@@ -133,26 +133,29 @@ public void windowClosing(WindowEvent arg0) {
...
+                    Language.INSTANCE.localize("instance.addmod"),
+                    Language.INSTANCE.localize("common.mod"),
+                    Language.INSTANCE.localize("common.add"),
+                    Language.INSTANCE.localize("instance.typeofmod"),
+                    modTypes,
+                    new String[] {"jar", "zip", "litemod"}
+                );
                 ArrayList<File> files = fcd.getChosenFiles();
-                if (files != null && files.size() >= 1) {
+                if (files != null && !files.isEmpty()) {
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/gui/dialogs/EditModsDialog.java]: 👍

File: src/main/java/com/atlauncher/mclauncher/LegacyMCLauncher.java (hunk 1 above)
Comment (dgelessus) [inline src/main/java/com/atlauncher/mclauncher/LegacyMCLauncher.java]: `URLClassLoader`'s `close` method was added in Java 7, so this causes `NoSuchMethodError`s on Java 6.

Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/mclauncher/LegacyMCLauncher.java]: I see, all good then :)

**Pull request merged**